tools/benchmark/src/main/java/com/datadog/benchmark/internal/reader/MemoryVitalReader.kt

```diff
@@ -7,16 +7,41 @@
 package com.datadog.benchmark.internal.reader
 
 import java.io.File
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
 
 /**
  * Reads the device's `VmRSS` based on the `/proc/self/status` file.
  * cf. documentation https://man7.org/linux/man-pages/man5/procfs.5.html
  */
 internal class MemoryVitalReader(
-    internal val statusFile: File = STATUS_FILE
+    internal val statusFile: File = STATUS_FILE,
+    private val readerDelay: Long = STATUS_FILE_READ_INTERVAL_MS
 ) : VitalReader {
 
-    override fun readVitalData(): Double? {
+    private var maxMemory: Double = 0.0
```

Comment:
you need to make this property thread-safe: write and read is performed from different threads

Reply:
Good point, I use AtomicLong to store the `maxMemory` data here

Reply:
probably simple `@Volatile` annotation will be enough, because you only get and set value